lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -1870,12 +1866,21 @@ static bool s1ap_send_enb_deregistered_ind(
     arg->current_ue_index =
         (uint8_t)(arg->handled_ues % S1AP_ITTI_UE_PER_DEREGISTER_MESSAGE);
 
-    // max ues reached
-    if (arg->current_ue_index == 0 && arg->handled_ues > 0) {
-      S1AP_ENB_DEREGISTERED_IND(arg->message_p).nb_ue_to_deregister =
-          S1AP_ITTI_UE_PER_DEREGISTER_MESSAGE;
+    // Max UEs reached for this ITTI message, send message to MME App
+    if (arg->handled_ues == arg->deregister_ue_count ||
+        (arg->current_ue_index == 0 && arg->handled_ues > 0)) {
```

Comment:
arg->handled_ues is always positive as it is incremented on line 1865.

Reply:
Right..! The condition was written based on earlier logic and was holding valid in current scenario as well, hence did not notice it. Thanks for suggestion. Removed this redundant check and updated the PR code.